auditing/auditing.go

```diff
@@ -11,6 +11,16 @@ import (
 type Config struct {
 	Component string
 	Log       *slog.Logger
+
+	// Async indexes audit traces asynchronously if set to true. if this functionality is implemented depends on the audit backend implementation.
+	// If this is set to true it can occur that audit traces get lost in case the backend is not available for receiving the trace.
+	Async bool
+	// AsyncRetry defines the amount of attempts to retry sending an audit trace to a backend in case it failed.
+	AsyncRetry int
+	// AsyncBackoff defines the backoff after a failed attempt to index an audit trace to a backend.
+	AsyncBackoff time.Duration
+	// AsyncTimeout sets a timeout for indexing a trace for the backend.
+	AsyncTimeout time.Duration
```

Comment:
These configs are declared for all auditing providers, but are only implemented for splunk.
Either these should be moved to the `SplunkConfig` or a different solution should be considered.

What if we'd create an `asyncAuditing` which would then only implement the async mechanism while forwarding the index requests to the actual/wrapped backend?

```golang
// would be used like this:
splunk, err := auditing.Splunk(...)
if err != nil {
  return nil, err
}
asyncAuditing := auditing.Async(auditing.AsyncConfig{Retry: 1, /*...*/}, splunk)
return asyncAuditing, nil
```